### PR TITLE
Support defining paginator on resource

### PR DESCRIPTION
**Pull request by coryb710**
Updates Pagination/Formatters to check the resource paginator instead of the default paginator. Currently, if you've set a paginator directly on your resource, it's ignored in favor of the default.

### CODE REVIEW

File: lib/jsonapi/utils/response/formatters.rb [hunk 1]
```diff
@@ -273,7 +273,7 @@ def result_options(records, options)
             end
 
             if JSONAPI.configuration.top_level_meta_include_page_count
-              data[:page_count] = page_count_for(data[:record_count])
+              data[:page_count] = apply_pagination?(options) ? page_count_for(data[:record_count]) : 1
```
Comment (coryb710) [inline lib/jsonapi/utils/response/formatters.rb]: If `JSONAPI.configuration.top_level_meta_include_page_count` is true but `apply_pagination?` is false, then page count should always equal 1

File: lib/jsonapi/utils/support/pagination.rb [hunk 2]
```diff
@@ -62,16 +62,21 @@ def record_count_for(records, options)
         #
         # @api private
         def paginator
-          @paginator ||= paginator_klass.new(page_params)
+          @paginator ||= begin
+            paginator_klass = "#{resource_paginator_name}_paginator".classify.constantize
```
Comment (coryb710) [inline lib/jsonapi/utils/support/pagination.rb]: I changed this from being its own method to living inside `paginator` because a) it's not used anywhere else and b) I thought it might confusing to have `paginator`, `paginator_klass`, and `resource_paginator_name` methods.

File: spec/controllers/users_controller_spec.rb [hunk 3]
```diff
@@ -101,7 +101,7 @@
     context 'with "page"' do
       context 'when using "paged" paginator' do
         before(:all) do
-          JSONAPI.configuration.default_paginator = :paged
```
Comment (coryb710) [inline spec/controllers/users_controller_spec.rb]: Need to change the resource paginator instead of the default config because of the way `@request.resource_klass._paginator` works in Pagination. If `@paginator` has not been set on the resource, then it is set to the default config paginator. That's all good for the first set of paginator tests, but when it moves on to the next ones and updates the default paginator setting, `@paginator` for UserResource is still set to what it was previously. 

So, in the course of running the "paged" tests, `UserResource._paginator` gets set to `:paged`, even if you never updated the UserResource directly & changed the default paginator setting. When it gets to the "offset" tests and you update the default paginator to "offset", then you'll get an error because `UserResource._paginator` is still set as `:paged`.

Basically, this change is solving an issue with the tests themselves, not an issue being raised by the code.

File: spec/jsonapi/utils/support/pagination_spec.rb [hunk 4]
```diff
@@ -77,7 +77,7 @@
     end
   end
 
-  describe '#count_pages_for' do
+  describe '#page_count_for' do
```
Comment (coryb710) [inline spec/jsonapi/utils/support/pagination_spec.rb]: Changed this to match the corresponding method name

**Comment by coryb710** [comment]
@phantomphildius Would be great if you could take a look as well!

**Review by palytoxin — COMMENTED** [review]
Thanks, this is very useful
